File: cfgs/java_cfgs_foo/HOW_TO_AVOID_OVERFLOW_IN_MODULAR_MULTIPLICATION.java
Nodes
1;special;;;
2;statement;long res = 0;14;
3;statement;a = foo_1 % foo_2;15;
4;conditional;foo_1 > 0;16;
5;conditional;(foo_1 % 2) == 1;17;
6;statement;res = (foo_1 + foo_2) % foo_3;18;
7;statement;a = (foo_1 * 2) % foo_2;20;
8;statement;b /= 2;21;
9;statement;return foo_1 % foo_2;23;
10;exit;;;
Edges
0;2;
1;3;
2;4;
3;9;5;
4;6;7;
5;7;
6;8;
7;4;
8;10;
9;

File: cfgs/pythoncfgs/HOW_TO_AVOID_OVERFLOW_IN_MODULAR_MULTIPLICATION.py
Nodes
1;statement;foo_1 = 0;8;
2;statement;foo_1 = foo_2 % foo_3;9;
3;conditional;foo_1 > 0;10;
4;statement;return foo_1 % foo_2;15;
5;conditional;foo_1 % 2 == 1;11;
6;statement;foo_1 = foo_2 * 2 % foo_3;13;
7;statement;foo_1 //= 2;14;
8;statement;foo_1 = (foo_2 + foo_3) % foo_4;12;
9;exit;;
Edges
0;2;
1;3;
2;4;5;
3;9;
4;6;8;
5;7;
6;3;
7;6;
8;
